src/client/transactions/mod.rs

```diff
@@ -242,20 +238,28 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store, A: TransactionAuthenticator> Client
             transaction_request.into(),
         )?;
 
-        // Check that the expected output notes is a subset of the transaction's output notes
-        let tx_note_ids: BTreeSet<NoteId> =
-            executed_transaction.output_notes().iter().map(|n| n.id()).collect();
+        // Check that the expected output notes matches the transaction outcome.
+        // We comprae authentication hashes where possible since that involves note IDs + metadata
+        // (as opposed to just note ID which remains the same regardless of metadata)
+        let tx_note_auth_hashes: BTreeSet<Digest> =
+            notes_from_output(executed_transaction.output_notes())
+                .map(Note::authentication_hash)
+                .collect();
 
         let missing_note_ids: Vec<NoteId> = output_notes
             .iter()
-            .filter_map(|n| (!tx_note_ids.contains(&n.id())).then_some(n.id()))
+            .filter_map(|n| {
+                (!tx_note_auth_hashes.contains(&n.authentication_hash())).then_some(n.id())
```

Comment:
shouldn't both the `contains` argument and the `then_some` arguments be the same?

```suggestion
                (!tx_note_auth_hashes.contains(&n.authentication_hash())).then_some(n.authentication_hash())
```